src/pdf_parser/pdf_utils/disambiguator/overlapping.py

```diff
@@ -0,0 +1,235 @@
+
+
+def reduce_overlapping_boxes_horizontal(
+    left_box: TextBlock,
+    right_box: TextBlock,
+    min_pixel_overlap_horizontal: Optional[int] = 5,
+) -> Tuple[TextBlock, TextBlock]:
+    """Reduce the size of overlapping boxes to eliminate overlaps in the horizontal axis."""
+    intersection_width = left_box.intersect(right_box).width
+    if intersection_width > min_pixel_overlap_horizontal:
+        if vertically_nested(left_box, right_box):
+            left_box = TextBlock(
+                block=Rectangle(
+                    x_1=left_box.coordinates[0],
+                    y_1=left_box.coordinates[1],
+                    x_2=left_box.coordinates[2] - intersection_width,
+                    y_2=left_box.coordinates[3],
+                ),
+                text=left_box.text,
+                id=left_box.id,
+                type=left_box.type,
+                parent=left_box.parent,
+                next=left_box.next,
+                score=left_box.score,
+            )
+        elif vertically_nested(right_box, left_box):
+            right_box = TextBlock(
+                block=Rectangle(
+                    x_1=right_box.coordinates[0] + intersection_width,
+                    y_1=right_box.coordinates[1],
+                    x_2=right_box.coordinates[2],
+                    y_2=right_box.coordinates[3],
+                ),
+                text=right_box.text,
+                id=right_box.id,
+                type=right_box.type,
+                parent=right_box.parent,
+                next=right_box.next,
+                score=right_box.score,
+            )
+        return left_box, right_box
+    return left_box, right_box
+
+
+def is_on_top(box1: TextBlock, box2: TextBlock) -> bool:
+    """Identify if box1 is on top of the bottom box."""
+    if box1.coordinates[1] > box2.coordinates[1]:
+        return True
+    return False
+
+
+def is_on_left(box1: TextBlock, box2: TextBlock) -> bool:
+    """Identify if box1 is on the left of the right box."""
+    if box1.coordinates[0] < box2.coordinates[0]:
+        return True
+    return False
+
+
+def vertically_nested(box1: TextBlock, box2: TextBlock) -> bool:
+    """
+    Identify if box1 is vertically nested in box2 or if box2 is vertically nested in box1.
+
+    I.e. (0,1,2,3) is vertically nested within (1,0,2,5) as the entire range of y values lie within the other box.
+    """
+    if (
+        box1.coordinates[1] > box2.coordinates[1]
+        and box1.coordinates[3] < box2.coordinates[3]
+        or box2.coordinates[1] > box1.coordinates[1]
+        and box2.coordinates[3] < box1.coordinates[3]
+    ):
+        return True
+    return False
+
+
+def horizontally_nested(box1: TextBlock, box2: TextBlock) -> bool:
+    """
+    Identify if box1 is horizontally nested in box2 or if box2 is horizontally nested in box1.
+
+    I.e. (2,0,4,100) is nested horizontally within (0,0,10,10) as the entire range of x values lie within the other box.
```

Comment:
A common theme to these functions is that the definitions aren't intuitive. i.e. I wouldn't know what "vertically nested" and "horizontally nested" mean as someone coming to the code base. I think the solution to this is to link to test cases with associated images in the docstring. Obviously, we don't have these yet, but one for the future. It's one of those tasks that is so geometric it would almost be good if visual docstrings existed!